### PR TITLE
All: Fix a flaky test case in crypto.test.ts

### PR DESCRIPTION
**Pull request by wh201906**
This PR fixes https://github.com/laurent22/joplin/pull/10696#issuecomment-2473695859.
Related CI run:
https://github.com/laurent22/joplin/actions/runs/11815017664/job/32915288005

### CODE REVIEW

File: packages/lib/services/e2ee/cryptoShared.ts [hunk 1]
```diff
@@ -36,12 +36,11 @@ export const generateNonce = async (nonce: Uint8Array) => {
...
 export const increaseNonce = async (nonce: Uint8Array) => {
-	const carry = 1;
 	const end = nonce.length - nonceCounterLength;
 	let i = nonce.length;
 	while (i-- > end) {
-		nonce[i] += carry;
-		if (nonce[i] !== 0 || carry !== 1) {
+		nonce[i] += 1;
+		if (nonce[i] !== 0) {
 			break;
```
Comment (laurent22) [inline packages/lib/services/e2ee/cryptoShared.ts:44]: Could this affect data that has already been encrypted with the new encryption method? From the name I'm guessing that not but please confirm. How is this function used?

Comment (wh201906) [reply packages/lib/services/e2ee/cryptoShared.ts:44]: This doesn't affect the encrypted data at all. The old implementation and and the new implementation generate exactly the same output with same input (nonce in this case). It's just a optimization.
This function is used when a new chunk of data is encrypted. Each trunk of data is encrypted with the same key and a different nonce, and this function is used to change the nonce.

Comment (laurent22) [reply packages/lib/services/e2ee/cryptoShared.ts:44]: Thanks for confirming!

**Comment by laurent22** [comment]
Thanks for checking this issue and fixing it @wh201906!

**Pull request merged**